# S.N.E.

In [1]:
#pip install spotipy

## Load libraries

In [1]:
import json
import spotipy
import pandas as pd
from spotipy.oauth2 import SpotifyClientCredentials

In [2]:
client_id = 'ca8a238d0af9478ab8898203a65bb8ec'
client_secret = 'a5a25b8983e841dcbed6a957aafb09a9'

client_credentials_manager = SpotifyClientCredentials(client_id, client_secret)
sp = spotipy.Spotify(client_credentials_manager=client_credentials_manager)

## Artist information

In [3]:
# get artist information
artist_info = 'https://open.spotify.com/artist/31esGCExk8cw4f990awEBP?si=58N-o_-0TCqm0Ed5mvFnDA&dl_branch=1'

artist = sp.artist(artist_info)

In [51]:
print('Artist name : ' + artist['name'])
print(artist['genres'])
print(artist['followers']['total'])
print(artist['popularity'])

Artist name : S.N.E
[]
118
19


SNE has 118 followers on Spotify and a popularity of 19 (popularity is rated from 0 to 100, with 0 being the lowest popularity score and 100 being the highest). He has no genre listed.

In [52]:
#artist

### Artist albums

In [7]:
result = sp.artist_albums(artist_info, album_type='album')

In [8]:
albums = result['items']

In [16]:
while result['next']:
    result = spotify.next(result)
    albums.extend(result['items'])
    
# print all album names
for album in albums:
    print(album['name'])  
    print(album['release_date'])
    print(album['total_tracks'])

Home Bound
2020-04-21
7


Sne only has 1 album on Spotify and it was released in April 2020. There are 7 tracks on it.

In [36]:
# print all album info
# for album in albums:
#     print(album)    

### What are the top tracks?

In [17]:
tracks = sp.artist_top_tracks(artist_info, country='ZA')

In [24]:
for track in tracks['tracks'][:5]:
    print('track : ' + track['name'])
    print('audio : ' + track['preview_url'])

track : Umlilo
audio : https://p.scdn.co/mp3-preview/d3253983007d80e0b50872aec79c4c033195812b?cid=ca8a238d0af9478ab8898203a65bb8ec
track : The Rain
audio : https://p.scdn.co/mp3-preview/4257ffeb3bd90222a039ca50f8f5a83a435360a2?cid=ca8a238d0af9478ab8898203a65bb8ec
track : Afternoon Drinks
audio : https://p.scdn.co/mp3-preview/2bf69c981d2b4e2f608158cbf6aeb170b0913d44?cid=ca8a238d0af9478ab8898203a65bb8ec
track : Home Bound
audio : https://p.scdn.co/mp3-preview/52b4a0fd3e7ef22ffbf3872ed1ddc39927d19021?cid=ca8a238d0af9478ab8898203a65bb8ec
track : Under the Covers
audio : https://p.scdn.co/mp3-preview/0fa95f7a6cd88d043f778e3c27ee533ded86a7ea?cid=ca8a238d0af9478ab8898203a65bb8ec


Get information about the top song (Umlilo).

In [58]:
sp.track('https://open.spotify.com/track/2mSfBkfI1YlYehRz6TNATF?si=756ce6ef8a374d44')

{'album': {'album_type': 'single',
  'artists': [{'external_urls': {'spotify': 'https://open.spotify.com/artist/31esGCExk8cw4f990awEBP'},
    'href': 'https://api.spotify.com/v1/artists/31esGCExk8cw4f990awEBP',
    'id': '31esGCExk8cw4f990awEBP',
    'name': 'S.N.E',
    'type': 'artist',
    'uri': 'spotify:artist:31esGCExk8cw4f990awEBP'}],
  'available_markets': ['AD',
   'AE',
   'AG',
   'AL',
   'AM',
   'AO',
   'AR',
   'AT',
   'AU',
   'AZ',
   'BA',
   'BB',
   'BD',
   'BE',
   'BF',
   'BG',
   'BH',
   'BI',
   'BJ',
   'BN',
   'BO',
   'BR',
   'BS',
   'BT',
   'BW',
   'BY',
   'BZ',
   'CA',
   'CH',
   'CI',
   'CL',
   'CM',
   'CO',
   'CR',
   'CV',
   'CW',
   'CY',
   'CZ',
   'DE',
   'DJ',
   'DK',
   'DM',
   'DO',
   'DZ',
   'EC',
   'EE',
   'EG',
   'ES',
   'FI',
   'FJ',
   'FM',
   'FR',
   'GA',
   'GB',
   'GD',
   'GE',
   'GH',
   'GM',
   'GN',
   'GQ',
   'GR',
   'GT',
   'GW',
   'GY',
   'HK',
   'HN',
   'HR',
   'HT',
   'HU',
   'ID',
   'I

### Who does Spotify consider to be related to SNE?

In [33]:
related = sp.artist_related_artists(artist_info)

In [34]:
related

{'artists': []}

Spotify does not recognise any artists related to SNE. Why?

In [65]:
sp.search('https://open.spotify.com/artist/31esGCExk8cw4f990awEBP?si=xMhXi-VNScyvQEHQyfaj7w&dl_branch=1')

{'tracks': {'href': 'https://api.spotify.com/v1/search?query=https%3A%2F%2Fopen.spotify.com%2Fartist%2F31esGCExk8cw4f990awEBP%3Fsi%3DxMhXi-VNScyvQEHQyfaj7w%26dl_branch%3D1&type=track&offset=0&limit=10',
  'items': [],
  'limit': 10,
  'next': None,
  'offset': 0,
  'previous': None,
  'total': 0}}

In [6]:
# collects 1,000 Track IDs and their associated track name, artist name, and popularity score.
artist_name = []
track_name = []
popularity = []
track_id = []

for i in range(0, 1000, 50):
    track_results = sp.search(q='year:2018', type='track', limit=50, offset=i)
    for i, t in enumerate(track_results['tracks']['items']):
        artist_name.append(t['artists'][0]['name'])
        track_name.append(t['name'])
        track_id.append(t['id'])
        popularity.append(t['popularity'])

In [7]:
a = sp.search(q='year:2018', type='track', limit=50)

In [8]:
track_dataframe = pd.DataFrame({'artist_name': artist_name, 'track_name': track_name, 'track_id': track_id, 'popularity': popularity})
print(track_dataframe.shape)
track_dataframe.head()

(1000, 4)


artist_name                                    track_name  \
0         Juice WRLD                        All Girls Are The Same   
1  Schoolgirl Byebye                                     Year,2015   
2         Juice WRLD                                  Lucid Dreams   
3      Fleetwood Mac  Rhiannon (Will You Ever Win) - 2018 Remaster   
4      Morgan Wallen                               Whiskey Glasses   

                 track_id  popularity  
0  4VXIryQMWpIdGgYR4TrjT1          83  
1  0UsmyJDsst2xhX1ZiFF3JW          25  
2  285pBltuF7vW8TeWk8hdRR          84  
3  4fbwTO3DJ2qryMddov9RbB          50  
4  6foY66mWZN0pSRjZ408c00          78

In [5]:
track_dataframe.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   artist_name  1000 non-null   object
 1   track_name   1000 non-null   object
 2   track_id     1000 non-null   object
 3   popularity   1000 non-null   int64 
dtypes: int64(1), object(3)
memory usage: 31.4+ KB


In [47]:
you_matter = sp.audio_features('https://open.spotify.com/track/2CIClzKLAcOJmqihyrtyrK?si=4028710c41e747b5')

In [17]:
you_matter_analysis = sp.audio_analysis('https://open.spotify.com/track/2CIClzKLAcOJmqihyrtyrK?si=4028710c41e747b5')

In [19]:
type(you_matter)

dict

In [48]:
you_matter

[{'danceability': 0.675,
  'energy': 0.55,
  'key': 11,
  'loudness': -9.171,
  'mode': 0,
  'speechiness': 0.0388,
  'acousticness': 0.56,
  'instrumentalness': 9.49e-06,
  'liveness': 0.102,
  'valence': 0.491,
  'tempo': 140.063,
  'type': 'audio_features',
  'id': '2CIClzKLAcOJmqihyrtyrK',
  'uri': 'spotify:track:2CIClzKLAcOJmqihyrtyrK',
  'track_href': 'https://api.spotify.com/v1/tracks/2CIClzKLAcOJmqihyrtyrK',
  'analysis_url': 'https://api.spotify.com/v1/audio-analysis/2CIClzKLAcOJmqihyrtyrK',
  'duration_ms': 210125,
  'time_signature': 4}]

In [50]:
you_matter_analysis

{'meta': {'analyzer_version': '4.0.0',
  'platform': 'Linux',
  'detailed_status': 'OK',
  'status_code': 0,
  'timestamp': 1610631450,
  'analysis_time': 46.23984,
  'input_process': 'libvorbisfile L+R 44100->22050'},
 'track': {'num_samples': 4633256,
  'duration': 210.12498,
  'sample_md5': '',
  'offset_seconds': 0,
  'window_seconds': 0,
  'analysis_sample_rate': 22050,
  'analysis_channels': 1,
  'end_of_fade_in': 0.0,
  'start_of_fade_out': 202.79729,
  'loudness': -9.171,
  'tempo': 140.063,
  'tempo_confidence': 0.438,
  'time_signature': 4,
  'time_signature_confidence': 0.968,
  'key': 11,
  'key_confidence': 0.74,
  'mode': 0,
  'mode_confidence': 0.642,
  'codestring': 'eJxVmwtyLDcOBK8yR-D_c_-LObM4kp43Yh0m1dNNgkChCqBPHfOuWsanfNau54w6xuf09jn3tL7rbZ-67vr0vkarkwfbXOUzR-2379k-bTd-OvYspeQ5fnHqnbvUUz6Dp_3Fmu2W-Rlzr8-addTVe_-M285nlMpXfHiWNj57tzr6bvMzK2_ec-529lqfec_4zLPObXzqs1xo27WWufm31df53NZ52eqVvfDbcedpt00ePnW5P981Dt-o7TN6Xa1MfnYmL-gdQ_D8-NxRjsOz26j1c32k9tJZNn-tLLJ8bin8qrBobLeG

In [ ]:
artist_name = []
track_name = []
popularity = []
track_id = []

for i in range(0, 1000, 50):
    track_results = sp.search(q='year:2018', type='track', limit=50, offset=i)
    for i, t in enumerate(track_results['tracks']['items']):
        artist_name.append(t['artists'][0]['name'])
        track_name.append(t['name'])
        track_id.append(t['id'])
        popularity.append(t['popularity'])